src/main/java/core/Game.java

```diff
@@ -284,6 +292,62 @@ public static void recordPlayerResults(TAGStatSummary[] statSummaries, Game game
         }
     }
 
+    /**
+     * The recommended way to run a game is via evaluations.Frontend, however that may not work on
+     * some games for some screen sizes due to the vagaries of Java Swing...
+     * <p>
+     * Test class used to run a specific game. The user must specify:
+     * 1. Action controller for GUI interactions / null for no visuals
+     * 2. Random seed for the game
+     * 3. Players for the game
+     * 4. Mode of running
+     * and then run this class.
+     */
+    public static void main(String[] args) {
```

Comment:
Mm I still don't like the main method being in the middle of this file.

Reply:
Which is why I deliberately moved it to the end as the last commit on the Pull Request!

Reply:
Added tool tips to the buttons to clarify. Show All shows the actions for all players when it is their turn (so that you can override their decision), while Show Self is the old default that only shows the actions available to the human player (and hence avoids information leakage),

Adding the extra information is not difficult in theory - but I'd rather not do it now as fiddling with the GUI layout tends to take longer than anticipated.